cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/draft/CodedException.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ *
+ */
+
+package io.cdap.cdap.datapipeline.draft;
```

Comment:
Does it make sense to put it into some common place and not inside `draft`? Looks like a useful abstraction.
Another option would be to reuse javax.ws.rs.ServerErrorException/ClientErrorException with it's Response.Status. May be we even can have a common exception handler, so that there is not need for own own code to convert it into `sendError`

Reply:
We have that in the CDAP platform itself, but not in the cdap-api that exposed to application. This draft management is implemented as part of the pipeline app, hence it doesn't have visibility to the internal CDAP platform classes.